app/controllers/users_controller.rb

```diff
@@ -14,57 +16,41 @@ def create
   end
 
   def show
-    @user = User.find(params[:id])
     @friendship = Friendship.find_by(user_id: @user.id, friend_id: current_user.id) || Friendship.find_by(user_id: current_user.id, friend_id: @user.id)
     if current_user.id == @user.id
       user_path(@user)
-    elsif current_user.friends.include?(@user) && @friendship.approved
-      user_path(@user)
-    elsif current_user.inverse_friends.include?(@user) && @friendship.approved
+    elsif current_user.all_friends.include?(@user) && @friendship.approved
       user_path(@user)
     else
       render "private"
     end
   end
 
-  def autocomplete_username
-    term = params[:q].downcase
-    @users = User.where("lower(username) LIKE '%#{term}%'")
-    respond_to do |format|
-      format.json {
-        render json: @users.map(&:username)
-      }
-    end
```

Comment:
The reason for having two of the same thing here was that with two autocomplete bars (each making their own AJAX requests), it won't work if they share the same method. At least, I could not find another way near the end of that ridiculous 40-hour day we spent on this. 

Did you test it after removing this method?